src/libxively/sft/xi_sft_logic.c

```diff
@@ -51,6 +60,10 @@ xi_state_t xi_sft_make_context( xi_sft_context_t** context,
     return state;
 
 err_handling:
+    if ( NULL != *context )
+    {
+        XI_SAFE_FREE( ( *context )->updateable_files_download_order );
+    }
```

Comment:
I think this isn't necessary here, is it?

Reply:
If there's an error in creating the context, the xi_sft_free_context is not called.  Therefore we should catch errors and remove any sort of allocations that were made in an attempt to build the context, no?  In this case this is only the first allocation in the context, but this is safer to do in case people add more allocations later, and it's harmless for now.

Reply:
I get confused by this, because I check from where can the execution jump to the err_handling label. And the only place is the XI_ALLOC_AT. In case the execution jumps from there then the context will be null. So this code never gets called. And there is countless code blocks we could introduce which won't be executed ever. So what's bugging me is why have you chosen this code from the infinite number of possible code blocks not to be executed ever.